### DISCUSSION TITLE
Ignore Working Tree Clean

### DISCUSSION THREAD

**Discussion by snayff** [community/community#25676] [Actions]
<p>I’ve got an Action that lints my code base, which works fine. However, when there is no change, say I update some data files, the Action fails as follows:</p>
<pre><code class="lang-auto">Run git config --global user.name 'nqp2'
  git config --global user.name 'nqp2'
  git config --global user.email 'snayff@users.noreply.github.com'
  git commit -am "ci lint and test"
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    pythonLocation: C:\hostedtoolcache\windows\Python\3.8.10\x64
On branch feature/update_gh_actions
Your branch is up to date with 'origin/feature/update_gh_actions'.

nothing to commit, working tree clean
Error: Process completed with exit code 1.
</code></pre>
<p>Is there a way to elegantly handle this, such that the Action passes when there is no action to take?</p>
<p>-Snayff</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You could check if there are changes using <a href="https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code"><code>git diff --exit-code</code></a> and make the commit only if there are any, e.g.:</p>
<pre><code class="lang-auto">git diff --exit-code || git commit -am "ci lint and test"
</code></pre>

**Comment by kingthorin** [comment]
<pre><code class="lang-bash">if ! git diff-index --quiet HEAD --; then 
          // Do whatever when there are changes
fi
</code></pre>
<p>Example: <a href="https://github.com/zaproxy/zap-extensions/blob/daace7400391da3218d8e213889c3cdb27b11861/.github/workflows/create_retirejs_update.yml" class="inline-onebox" rel="noopener nofollow ugc">zap-extensions/create_retirejs_update.yml at daace7400391da3218d8e213889c3cdb27b11861 · zaproxy/zap-extensions · GitHub</a></p>
